scripts/modernize.sh

```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+# Runs the Clang Modernizer in parallel on the code base.
+# Requires a compilation database in the build directory.
+
+git ls-files '*.cpp' | xargs -I{} -P $(nproc) clang-modernize -p build -final-syntax-check -format -style=file -summary -for-compilers=clang-3.4,gcc-4.8 -include . -exclude third_party {}
```

Comment:
The reason we have to pipe the `.cpp` file into it is that the compilation database generated by CMake only contains the compiler commands for a single TU. That is, you have commands in there like `g++ main.cc -o main`, which only match the `.cpp` files, since the headers are included by the preprocessor.

That said, the tool *does* work on headers we just don't have to specify them, check the initial run here:
https://github.com/Project-OSRM/osrm-backend/pull/1603/files